### PR TITLE
 tot uses prow config to determine fallback GCS path 

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/7056

I also wrote a cli tool that checks all the expected GCS paths that need to exist given a live prow deployment and a GCS bucket.

/cc @stevekuznetsov @rmmh @BenTheElder 
/area prow

### CODE REVIEW

File: prow/cmd/tot/main.go [hunk 1]
```diff
@@ -192,9 +221,55 @@ func (f fallbackHandler) get(jobName string) int {
...
+				break
+			}
+		}
+	}
+	// If spec is still nil, we know nothing about the requested job.
+	if spec == nil {
+		logrus.Errorf("requested job is unknown to prow: %s", jobName)
+		return ""
+	}
+	return fmt.Sprintf("%s/%s", strings.TrimSuffix(f.bucket, "/"), gcs.LatestBuildForSpec(spec))
```
Comment (kargakis) [inline prow/cmd/tot/main.go]: I don't feel strong about trimming trailing slashes here - I can invalidate `--fallback-bucket` invocations that include it is another option.

**Comment by kargakis** [comment]
Added a cli tool that checks all the expected GCS paths that need to exist given a live prow deployment and a GCS bucket.

**Comment by stevekuznetsov** [comment]
CLI tool seems reasonable to me

/cc @BenTheElder @rmmh

**Comment by BenTheElder** [comment]
This seems good to me.
/lint

File: prow/cmd/tot/main.go [hunk 2]
```diff
@@ -51,13 +59,28 @@ func gatherOptions() options {
...
+		"Fallback to top-level bucket for jobs that lack a last vended build number. The bucket layout is expected to follow https://github.com/kubernetes/test-infra/tree/master/gubernator#gcs-bucket-layout",
 	)
 
 	flag.Parse()
 	return o
 }
 
+func (o *options) Validate() error {
+	if o.configPath != "" && o.fallbackBucket == "" {
+		return errors.New("you need to provide a bucket to fallback to when the prow config is specified")
```
[the remark this answers is not in the document]
Comment (kargakis) [reply prow/cmd/tot/main.go]: No, these two go together. Mutual exclusive to them is providing a url template but if I invalidate such a combination I need to empty the default from the flag which is a breaking change.

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply prow/cmd/tot/main.go]: Oh, yuck. I can't read. Disregard.

**Pull request merged**